CabalHelper/Compile.hs

```diff
@@ -413,6 +413,9 @@ errorInstallCabal cabalVer _distdir = panic $ printf "\
  where
    sver = showVersion cabalVer
 
+helperPrepared :: Version -> IO Bool
+helperPrepared compCabalVersion = isJust <$> cachedExe compCabalVersion
```

Comment:
Indeed it would but ghc-mod caches cabal-helper info and the cache only gets invalidated when the `cabal` file changes (well `dist/setup-config` really). This is not ideal since Cabal itself only recompiles `Setup.hs` when the source module changed but that's just how it is right now.